logstash-core/lib/logstash/agent.rb

```diff
@@ -10,80 +10,130 @@
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
-  attr_reader :logger
+  attr_reader :logger, :state
 
-  def initialize(logger)
-    @logger = logger
+  def initialize(params)
```

Comment:
I'm not a fan of having this global `params` argument but different agents can have different sources of events...
any suggestions here @andrewvc ?